tensorflow_quantum/core/serialize/serializer.py

```diff
@@ -434,6 +536,20 @@ def serialize_circuit(circuit_inp):
                    old_moment.operations))
         circuit[moment_ind] = new_moment
 
+    # Demote cirq.controlled_operations (controlled gates) to their sub_gate
+    # types with _tfq_control_qubits and _tfq_control_values fields so that
+    # the gates can still get picked up by the serializer. There would be no way
+    # to discern controlledgates from one another otherwise.
+    for i, moment in enumerate(circuit):
+        for op in moment:
+            if isinstance(op,
+                          cirq.ops.controlled_operation.ControlledOperation):
+                tfq_compatible = op.sub_operation
```

Comment:
I thought maybe recursion of controlled operations could be a problem here, but I tried the following snippet:
```
q0, q1, q2 = cirq.GridQubit.rect(1, 3)
cx = cirq.X(q0).controlled_by(q1)
ccx = cirq.ops.ControlledOperation((q2,), cx)
```
and it turns out `ccx.sub_operation` is still an X gate.  Could you add a comment here, maybe with a code snippet like this, just to clarify that it is not possible to naturally create a sub operation that is Controlled?

Reply:
This is a feature of the Cirq library so I don't think we really need to make that clear here.